store/write_controlled_store.go

```diff
@@ -34,19 +40,15 @@ func (w *WriteControlledStore) acquireSyncRef(id imap.InternalMessageID) *syncRe
 
 	v, ok := w.entryTable[id]
 	if !ok {
-		var s *syncRef
-
-		if len(w.lockPool) != 0 {
-			s = w.lockPool[0]
-			s.counter = 1
-			w.lockPool = w.lockPool[1:]
-		} else {
-			s = &syncRef{counter: 1}
+		v, ok := w.lockPool.Get().(*syncRef)
+		if !ok {
+			panic("Invalid Type Cast")
```

Comment:
```suggestion
			panic("invalid type cast")
```

little style nitpick, go panics should all be lowercase (for some reason)

Reply:
Done